### PR TITLE
ddnet: new recipe

### PR DESCRIPTION
**Pull request by n0toose**
This recipe was largely based on preexisting work for Teeworlds. All of the Haiku-specific patches were upstreamed.

The maintainers are also super nice.

---

Fixes https://github.com/haikuports/haikuports/issues/7914, I added @threedeyes as a co-author because this is largely based on previous work.

---

#### Screenshots

image

image

image

### CODE REVIEW

**Comment by n0toose** [comment]
P.S. I'm getting some weird input issue in my VM which makes the game hard to play, this probably isn't reproducible in other setups but thought I'd share.

File: games-action/ddnet/ddnet-2023.03.09.recipe [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	devel:libz$secondaryArchSuffix
+	"
+
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:cargo$secondaryArchSuffix >= 0.61.0
+	cmd:make
+	cmd:ninja
+	"
```
Comment (korli) [inline games-action/ddnet/ddnet-2023.03.09.recipe:78]:
```suggestion
BUILD_PREREQUIRES="
	cmd:cargo$secondaryArchSuffix >= 0.61.0
	cmd:cmake
	cmd:gcc$secondaryArchSuffix
	cmd:make
	cmd:ninja
	"
```

File: games-action/ddnet/ddnet-2023.03.09.recipe [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+BUILD()
+{
+	mkdir -p build
+	cd build
+	cmake .. \
```
Comment (korli) [inline games-action/ddnet/ddnet-2023.03.09.recipe:83]:
```suggestion
cmake -B build -S . \
```

File: games-action/ddnet/ddnet-2023.03.09.recipe [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		-DCLIENT=ON \
+		-DVULKAN=OFF \
+		-GNinja
+	ninja $jobArgs
+}
+
+INSTALL()
+{
+	cd build
+	ninja install
```
Comment (korli) [inline games-action/ddnet/ddnet-2023.03.09.recipe:95]:
```suggestion
	ninja -C build install
```

**Comment by n0toose** [comment]
Note: I thought about adding these two changes, but thought it was best to leave the Teeworlds recipe as unchanged as possible so as to speed up the review process/because I thought it'd be more acceptable that way because of some sort of a de facto rule I may have missed. It may be a good idea to "port" these over to the Teeworlds recipe too.

File: games-action/ddnet/ddnet-2023.03.09.recipe [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+DDRace, a Teeworlds modification with a unique cooperative gameplay. Help \
+each other play through custom maps with up to 64 players, compete against \
+the best in international tournaments, design your own maps, or run your own \
+server. The official servers are around the world. All ranks made on official \
+servers are available worldwide and you can collect points!"
+HOMEPAGE="https://ddnet.org"
+COPYRIGHT="2007-2014 Magnus Auvinen \
+	2010-2011 Shereef Marzouk \
+	2013-2022 Dennis Felsing"
+LICENSE="Zlib"
```
Comment (korli) [inline games-action/ddnet/ddnet-2023.03.09.recipe:12]:
```suggestion
LICENSE="Zlib
	CC-BY-SA-3.0
	SIL Open Font License v1.1"
```

Please add a copy of the license CC-BY-SA-3.0 like with https://github.com/haikuports/haikuports/blob/2490464cd13f580060acebc18ef617f55e30da44/games-action/lugaru/licenses/CC-BY-SA-3.0

**Comment by n0toose** [comment]
Made a typo I forgot to track before pushing, sec.

**Comment by n0toose** [comment]
Sorry about that, was essentially trying to resolve the issues in real time.

File: games-action/ddnet/ddnet-2023.03.09.recipe [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:ninja
+	"
+
+BUILD()
+{
+cmake -B build -S . \
+		-DCMAKE_INSTALL_PREFIX=$appsDir/DDNet \
```
Comment (Begasus) [inline games-action/ddnet/ddnet-2023.03.09.recipe:84]: ddnet uses GNUInstallDirs: https://github.com/ddnet/ddnet/blob/master/CMakeLists.txt#L3020

Maybe worth to use $cmakeDirArgs

File: games-action/ddnet/ddnet-2023.03.09.recipe [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+	devel:libswresample$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cargo$secondaryArchSuffix >= 0.61.0
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:ninja
+	"
```
Comment (Begasus) [inline games-action/ddnet/ddnet-2023.03.09.recipe:79]: Missing cmd:python3
```diff
+	cmd:python3
```

Comment (n0toose) [reply games-action/ddnet/ddnet-2023.03.09.recipe:79]: Would've probably never noticed that one, what is it needed for?

File: games-action/ddnet/ddnet-2023.03.09.recipe (hunk 5 above)
Comment (n0toose) [reply games-action/ddnet/ddnet-2023.03.09.recipe:84]: Doesn't exactly work.

**Comment by n0toose** [comment]
Disabled 32-bit according to @Begasus's instructions, "it compiles but crashes on start".

**Pull request merged**